forem/forem
# Takeover code for `publishing_from_rss` section in Settings

This may not be the final list of to-do's. We will update this list as we discover more: 
- [x] Update the Publishing from RSS options in `settings/extensions` tab to use the new `users_settings` table 
- [ ] Check for other instances where the affected fields are being used

## What type of PR is this? (check all applicable)


## Description
In this PR, I point all changes within the Publishing from RSS section in `settings/extensions` to the `users_settings` table. Changes previously pointed to the `users` table.
On advice from @rhymes, I've decided to leave the `feed_fetched_at` field in `users` table. The plan was to migrate the `feed_fetched_at` data from `users` to `users_settings`, but I'm holding off on this migration to ensure that nothing unforeseen gets broken.
This PR shall be merged into [PR 13904](https://github.com/forem/forem/pull/13904).


## QA Instructions, Screenshots, Recordings
1. Sign into `localhost`; in rails console, pull up the same user that you are signed in as.
2. In rails console, run `user.setting` and take note the values for `feed_mark_canonical`, `feed_referential_link`, `feed_url`. Also call these attributes on your user, and note that their values match those in the `setting` record. 
3. In the app, navigate to `localhost:xxxx/settings/extensions`; make some changes under the Publishing from RSS section. Click Submit/Save Connect Settings.
4.  In the console, reload your user (`user.reload`) and re-run `user.setting`. Now the relevant `setting` values should differ from the `user.` values, as the `users` record was left unchanged, but the `settings` record was changed accordingly.
5. In the console, run `articles.destroy_all` on your user. In the app, navigate to `/dashboard` and ensure that there are no posts. Back in `/settings/extensions`, click the "Fetch Feed Now" button. The "Last Fetched:" value should be updated to the current date and time. Back in `/dashboard`, you should see some posts in draft status.


### UI accessibility concerns? NONE (backend changes)

## Added tests?

      have not been included_
- [X] Not yet; will write after review of the code implementation

## [Forem core team only] How will this change be communicated?
